src/orange/surf/GeneralQuadric.hh

```diff
@@ -41,9 +41,9 @@ class GeneralQuadric
     //@{
     //! Type aliases
     using Intersections = Array<real_type, 2>;
-    using StorageSpan = Span<const real_type, 10>;
+    using StorageSpan = LdgSpan<const real_type, 10>;
     using Storage = StorageSpan;  // DEPRECATED
-    using SpanConstReal3 = Span<const real_type, 3>;
+    using SpanConstReal3 = LdgSpan<const real_type, 3>;
```

Comment:
Ah, this could be a problem! This is the return value from *local* data (registers/stack) that's already been loaded into the `GeneralQuadratic` object. Only the `StorageSpan` should be changed here. (I don't think it's *used* on device but it would cause a runtime error I think.)

Reply:
Right, I don't know if it would be a runtime error or if the compiler figures out that __ldg is applied to a register variable and ignores it but it doesn't make sense to use LdgSpan here.

Reply:
Thinking about more, the compiler can't always find out which memory space the pointer belongs to. I don't think we should even change `StorageSpan`; `GeneralQuadric::data` returns a `StorageSpan` which would also result in a runtime error since the data is already loaded in registers. 
Similarly, the constructor `GeneralQuadric(Storage);` would restrict constructing a surface from a Span pointing to global memory data only. We can change the constructor to accept either `Span` or `LdgSpan` and users should wrap the `Span` returned by `data`, `first`, ... if they know the object resides in GMem.